doc/src/manual/embedding.md

```diff
@@ -284,6 +284,50 @@ args[1] = some_other_value;
 JL_GC_POP();
 ```
 
+Each scope must have only one call to `JL_GC_PUSH*`. Hence, if it is necessary to protect more
+variables inside the same function, one can use inner blocks:
+
+```c
+jl_value_t *ret1 = jl_eval_string("sqrt(2.0)");
+JL_GC_PUSH1(&ret1);
+jl_value_t *ret2 = 0;
+{
+    jl_function_t *func = jl_get_function(jl_base_module, "exp");
+    ret2 = jl_call1(func, ret1);
+    JL_GC_PUSH1(&ret2);
+    // Do something with ret2.
+    JL_GC_POP();    // This pops ret2.
+}
+JL_GC_POP();    // This pops ret1.
+```
+
+If it is required to hold the pointer to a variable between functions (or block scopes), then it is
+not possible to use `JL_GC_PUSH*`. In this case, it is necessary to create and keep a reference to the
+variable in the Julia global scope. One simple way to accomplish this is to create a global `IdDict`
+that will hold the references, avoiding deallocation by the GC.
+
+```c
+// After initialization, create a global `IdDict` called `refs`.
+jl_eval_string("refs = IdDict()");
+
+...
+
+// `var` is the variable we want to protect between function calls.
+jl_value_t* var = 0;
+
+...
+
+var = jl_eval_string("sqrt(2.0)");
+
+// To protect `var`, add its reference to `refs`.
+jl_value_t* refs = jl_eval_string("refs");
+jl_function_t* setindex = jl_get_function(jl_base_module, "setindex!");
+jl_call3(setindex, refs, var, var);
+```
+
+The GC can be allowed to deallocated the variables by removing the reference to them in `refs` using
+the function `delete!`.
```

Comment:
Be careful with duplicated objects. (You can hold multiple references to the same object)

Reply:
ok!